### PR TITLE
feat: support RoboTwin 2.0 with new interface

### PR DESCRIPTION
**Pull request by guozhen1997**
### Description
Support RoboTwin 2.0 with new interface

- Support robotwin vector env;
- Support official implement version of OpenVLA-oft.
- Fix lora weight save.
- Add OpenVLA-oft save helper.
- Fix maniskill offload.

### How has this been tested?

place_empty_cup
image

beat_block_hammer
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/embodiment/config/robotwin_place_empyt_cup_ppo_openvlaoft.yaml [hunk 1]
```diff
@@ -113,26 +113,31 @@ actor:
...
   micro_batch_size: 1 # 1
-  global_batch_size: 40 # 640
+  global_batch_size: 32 # 640
   seed: 1234
-  enable_offload: False
+  enable_offload: True
 
   # Override the default values in model/openvla_oft
   model:
     model_path: "/mnt/public/wph/models/robotwin2_place_empty_cup_seed1k_sft_aloha_25chunks_10k/"
```
Comment (zoeyuchao) [inline examples/embodiment/config/robotwin_place_empyt_cup_ppo_openvlaoft.yaml]: same

File: .gitignore [hunk 2]
```diff
@@ -17,4 +17,5 @@ outputs/*
 .venv
 .docs-venv
 uv.lock
-rlinf/envs/maniskill/assets
\ No newline at end of file
+rlinf/envs/maniskill/assets
+*.mp4
```
Comment (XuS1994) [inline .gitignore]: *.mp4 forget to delete

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+    - End-effector 3D position control (x, y, z): 3 dimensions
+    - 3D rotation control (roll, pitch, yaw): 3 dimensions
+    - Gripper control (open/close): 1 dimension
+    - Joint position control: 7 dimensions
+    - **Total**: 14 dimensions
+
+Dependency Installation
+-----------------------
+
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/robotwin.rst:145]: Missing RLinf clone like in other examples.

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+RoboTwin Assets are asset files required by the RoboTwin environment and need to be downloaded from HuggingFace.
+
+.. code-block:: bash
+
+   # 1. Clone RoboTwin repository
+   git clone https://github.com/RoboTwin-Platform/RoboTwin.git
+   cd RoboTwin
+   
+   # 2. Switch to RLinf_support branch
+   git checkout RLinf_support
```
Comment (XuS1994) [inline docs/source-en/rst_source/examples/robotwin.rst]: Would it be easier to maintain by forking a RoboTwin repo with RLinf, rather than using their branch?

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 5]
```diff
@@ -0,0 +1,353 @@
+RL with RoboTwin Simulator
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/robotwin.rst]: Simulator -> Benchmark.

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+.. code-block:: yaml
+
+   actor:
+     model:
+       model_path: "/path/to/RLinf-OpenVLAOFT-RoboTwin-SFT-place_empty_cup"  # SFT model path
+       model_type: "openvla_oft"                                             # Model type set to openvla_oft
+       implement_version: "offical"                                          # openvla_oft implementation version (RLinf OpenVLA-OFT model implementation integrates the official OFT version and RLinf SFT fine-tuned version, RoboTwin environment uses the official version)
+       action_dim: 14                                                        # RoboTwin action dimension (14D)
+       use_proprio: True                                                     # Whether to use proprioception information
+       proprio_dim: 14                                                       # Proprioception dimension (must match action_dim)
```
Comment (XuS1994) [inline docs/source-en/rst_source/examples/robotwin.rst]: If proprio dim must match action_dim, we can use action_dim directly

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+RLinf provides a pre-configured RoboTwin environment Docker image that includes all required dependencies and can be used directly, **skipping all subsequent installation steps**.
+
+.. code:: bash
+
+   docker run -it --rm --gpus all \
+      --shm-size 20g \
+      --network host \
+      --name rlinf \
+      -v .:/workspace/RLinf \
+      rlinf/rlinf:agentic-rlinf0.1-robotwin-openvlaoft-openpi
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/robotwin.rst]: rlinf/rlinf:agentic-rlinf0.1-robotwin?

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+   - RLinf RoboTwin environment dependencies
+   - Compatibility patches applied
+   - Support for OpenVLA-OFT, OpenPI models
+
+   **After using the Docker image, you can directly proceed to the** `RoboTwin Repository Clone and Assets Download`_ **, ** `Model Download`_ **and** `Running Scripts`_ **sections, skipping all subsequent installation steps.**
+
+Method 2: Manual Installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``requirements/install.sh`` script with the ``--env robotwin`` parameter to install the RoboTwin environment. Replace the ``--model openvla-oft`` parameter with the corresponding model name (``openvla``, ``openvla-oft``, or ``openpi``) based on the model you want to train:
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/robotwin.rst]: openvla is not supported yet?

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+
+RoboTwin Repository Clone and Assets Download
+---------------------------------------------
+
+RoboTwin Assets are asset files required by the RoboTwin environment and need to be downloaded from HuggingFace.
+
+.. code-block:: bash
+
+   # 1. Clone RoboTwin repository
+   git clone https://github.com/RoboTwin-Platform/RoboTwin.git
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/robotwin.rst]: git clone https://github.com/RoboTwin-Platform/RoboTwin.git -b RLinf_support. Remove the checkout later

File: docs/source-en/rst_source/examples/robotwin.rst [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+Model Download
+-----------------------
+
+Before starting training, you need to download the corresponding SFT model:
+
+.. code-block:: bash
+
+   # Download OpenVLA-OFT model
+   pip install huggingface-hub
+   huggingface-cli download RLinf/RLinf-OpenVLAOFT-RoboTwin-SFT-place_empty_cup
```
Comment (andylin-hao) [inline docs/source-en/rst_source/examples/robotwin.rst]: Not aligned with other examples. Missing the git clone approach and HF mirror setting

File: examples/embodiment/config/model/openvla_oft.yaml [hunk 11]
```diff
@@ -1,5 +1,5 @@
 model_type: "openvla_oft"
-implement_version: "rlinf"
+implement_version: "rlinf" # supported versions: ['rlinf', 'offical']
```
Comment (XuS1994) [inline examples/embodiment/config/model/openvla_oft.yaml:2]: what is the diff of rlinf and offical?
And if openpi model also has offical selection?

File: rlinf/models/embodiment/mlp_policy/mlp_policy.py [hunk 12]
```diff
@@ -103,16 +103,16 @@ def preprocess_env_obs(self, env_obs):
...
+        elif forward_type == ForwardType.SAC_Q:
             return self.sac_q_forward(**kwargs)
-        elif forward_type == "crossq_forward":
+        elif forward_type == ForwardType.CROSSQ:
             return self.crossq_forward(**kwargs)
-        elif forward_type == "crossq_q_forward":
+        elif forward_type == ForwardType.CROSSQ_Q:
             return self.crossq_q_forward(**kwargs)
-        elif forward_type == "default_forward":
+        elif forward_type == ForwardType.DEFAULT:
```
Comment (XuS1994) [inline rlinf/models/embodiment/mlp_policy/mlp_policy.py:115]: a bit confused, what exactly does DEFAULT refer to?

File: rlinf/models/embodiment/openvla_oft/openvla_utils.py [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+            proprio,
+        ),
+        a_min=-1.0,
+        a_max=1.0,
+    )
+
+    return normalized_proprio
+
+
+def find_checkpoint_file(pretrained_checkpoint, file_pattern):
```
Comment (XuS1994) [inline rlinf/models/embodiment/openvla_oft/openvla_utils.py:67]: `find_checkpoint_file` seem to be used as a normal util. So it can be moved to rlinf/models/utils.py

File: rlinf/models/embodiment/openvla_oft/openvla_utils.py [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+    vla.model.vision_backbone.load_state_dict(state_dict)
+
+    # Use the model component instead of wrapper and convert to bfloat16
+    vla = vla.model
+    vla.vision_backbone = vla.vision_backbone.to(torch.bfloat16)
+
+    return vla
+
+
+def load_dataset_stats(cfg: DictConfig) -> None:
```
Comment (XuS1994) [inline rlinf/models/embodiment/openvla_oft/openvla_utils.py:178]: `load_dataset_stats` also seem to be used as a normal util. So it can be moved to rlinf/models/utils.py

File: rlinf/workers/actor/fsdp_sac_policy_worker.py [hunk 15]
```diff
@@ -22,12 +22,13 @@
 import torch.nn.functional as F
 from omegaconf import DictConfig
 
-# import rlinf.algorithms.advantages_sac  # noqa: F401
```
Comment (XuS1994) [inline rlinf/workers/actor/fsdp_sac_policy_worker.py:25]: Will deleting the import directly affect the usage of SAC?

File: docs/source-en/rst_source/examples/robotwin.rst (hunk 6 above)
Comment (guozhen1997) [reply docs/source-en/rst_source/examples/robotwin.rst]: The `proprio_dim` is used by the proprio projector to encode environment state information, while `action_dim` is used by the action head as a control-space constraint. In RoboTwin, these two dimensions happen to be the same, but they serve different roles conceptually. I will remove the comment ` (must match action_dim)`.

File: docs/source-en/rst_source/examples/robotwin.rst (hunk 4 above)
Comment (guozhen1997) [reply docs/source-en/rst_source/examples/robotwin.rst]: This is a collaboration, and there will be interactions between both communities.

File: rlinf/models/embodiment/mlp_policy/mlp_policy.py (hunk 12 above)
Comment (guozhen1997) [reply rlinf/models/embodiment/mlp_policy/mlp_policy.py:115]: ForwardType.DEFAULT refers to the original forward function. In our current implementation, the forward function is used as a router to dispatch to different forward implementations.

File: rlinf/models/embodiment/openvla_oft/openvla_utils.py (hunk 13 above)
Comment (guozhen1997) [reply rlinf/models/embodiment/openvla_oft/openvla_utils.py:67]: For now, this function is only used for openvla-oft model to load proprio_projector and vision_backbone modules.

File: rlinf/models/embodiment/openvla_oft/openvla_utils.py (hunk 14 above)
Comment (guozhen1997) [reply rlinf/models/embodiment/openvla_oft/openvla_utils.py:178]: This function is only used by the openvla-oft model to load normalization statistics, since different models handle their normalization stats in different ways.

File: rlinf/workers/actor/fsdp_sac_policy_worker.py (hunk 15 above)
Comment (guozhen1997) [reply rlinf/workers/actor/fsdp_sac_policy_worker.py:25]: This is an unused comment.

File: examples/embodiment/config/model/openvla_oft.yaml (hunk 11 above)
Comment (guozhen1997) [reply examples/embodiment/config/model/openvla_oft.yaml:2]: rlinf: refers to the OpenVLA-OFT model released on Hugging Face by RLinf, trained using RLinf's modified OpenVLA-OFT repository.
official: refers to models trained from the official OFT repository, which support use_film and use_proprio.
I will also add these comments in model config yaml.

File: .gitignore (hunk 2 above)
Comment (guozhen1997) [reply .gitignore]: removed
